src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,656 @@
+package org.kohsuke.github;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Represents a repository rule.
+ */
+public class GHRepositoryRule {
+    private Type type;
+    private RulesetSourceType rulesetSourceType;
+    private String rulesetSource;
+    private long rulesetId;
+    private Map<String, JsonNode> parameters;
+
+    /**
+     * Gets the type.
+     *
+     * @return the type
+     */
+    public Type getType() {
+        return this.type;
+    }
+
+    /**
+     * Gets the ruleset source type.
+     *
+     * @return the ruleset source type
+     */
+    public RulesetSourceType getRulesetSourceType() {
+        return this.rulesetSourceType;
+    }
+
+    /**
+     * Gets the ruleset source.
+     *
+     * @return the ruleset source
+     */
+    public String getRulesetSource() {
+        return this.rulesetSource;
+    }
+
+    /**
+     * Gets the ruleset id.
+     *
+     * @return the ruleset id
+     */
+    public long getRulesetId() {
+        return this.rulesetId;
+    }
+
+    /**
+     * Gets a parameter. ({@link GHRepositoryRule.Parameters Parameters} provides a list of available parameters.)
+     *
+     * @param parameter
+     *            the parameter
+     * @param <T>
+     *            the type of the parameter
+     * @return the parameters
+     */
+    public <T> Optional<T> getParameter(Parameter<T> parameter) {
+        return Optional.ofNullable(this.parameters).map(p -> p.get(parameter.getKey())).map(parameter);
+    }
+
+    /**
+     * The type of the ruleset.
+     */
+    public static enum Type {
```

Comment:
Add `UNKNOWN`, make the `type` field a `String`, and use the EnumUtils to convert in the getter.  See other enums with `UNKNOWN` for examples.

Reply:
Not sure, if I get your idea, but do you want me to add `UNKNOWN` in case of the API returning something the client does not understand?

Reply:
Yes, like this.  We've seen multiple cases of GitHub adding new types and we don't want this library to throw when it encounters them. 
https://github.com/hub4j/github-api/blob/cab9bbb8cb37599362cb37f0c278ab262298c59b/src/main/java/org/kohsuke/github/GHCheckRun.java#L102-L122

Reply:
NOTE: we don't do this for all enums but this one seems like on a case where there are likely to be new parameter types sooner or later.

Reply:
got it.